numpy/core/src/multiarray/refcount.c

```diff
@@ -51,6 +51,31 @@ PyArray_Item_INCREF(char *data, PyArray_Descr *descr)
             PyArray_Item_INCREF(data + offset, new);
         }
     }
+    else if (PyDataType_HASSUBARRAY(descr)) {
+        int size, i, inner_elsize;
```

Comment:
Should any of these be `intp`?

Reply:
I think `int` is correct, since that is what dtypes use for the `elsize` so we are stuck with it.

Reply:
I guess we could use `intp` anyway...